src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_pending_resumption_handler.cc

```diff
@@ -31,258 +31,281 @@
+    LOG4CXX_DEBUG(logger_, "No pending resumptions");
+    return;
+  }
+  auto& pending_resumption = pending_requests_.front();
+  if (pending_resumption.waiting_for_hmi_response_) {
+    LOG4CXX_DEBUG(logger_,
+                  "Pending resumption for  "
+                      << pending_resumption.app_id_
+                      << " is already waiting for HMI response");
+    return;
+  }
+  SendHMIRequestForNotSubscribed(pending_resumption);
+  pending_resumption.waiting_for_hmi_response_ = true;
 }
 
-std::map<std::string, bool>
-VehicleInfoPendingResumptionHandler::ExtractSubscribeResults(
-    const smart_objects::SmartObject& response,
-    const smart_objects::SmartObject& request) const {
+void VehicleInfoPendingResumptionHandler::on_event(
+    const application_manager::event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  sync_primitives::AutoLock lock(lock_);
   using namespace application_manager;
-  const hmi_apis::Common_Result::eType result_code =
-      static_cast<hmi_apis::Common_Result::eType>(
-          response[strings::params][application_manager::hmi_response::code]
-              .asInt());
-  bool succesfull_response = (result_code == hmi_apis::Common_Result::SUCCESS ||
-                              result_code == hmi_apis::Common_Result::WARNINGS);
-  const auto response_keys =
-      response[application_manager::strings::msg_params].enumerate();
-  const auto request_keys =
-      request[application_manager::strings::msg_params].enumerate();
-
-  auto response_params = response[strings::msg_params];
-
-  std::map<std::string, bool> subscription_results;
-
-  if (!succesfull_response) {
-    for (auto key : request_keys) {
-      subscription_results[key] = false;
-    }
+  if (pending_requests_.empty()) {
+    LOG4CXX_DEBUG(logger_, "Not waiting for any response");
+    return;
   }
 
-  if (succesfull_response) {
-    for (auto key : request_keys) {
-      if (!helpers::in_range(response_keys, key)) {
-        subscription_results[key] = true;
-      } else {
-        const auto kSuccess =
-            hmi_apis::Common_VehicleDataResultCode::VDRC_SUCCESS;
-        const auto vd_result_code =
-            response_params[key][application_manager::strings::result_code]
-                .asInt();
-        subscription_results[key] = vd_result_code == kSuccess;
-      }
-    }
+  auto current_pending = pending_requests_.front();
+
+  auto response_message = event.smart_object();
+  smart_objects::SmartObject converted_msg_params(
+      response_message[strings::msg_params]);
+
+  custom_vehicle_data_manager_.CreateMobileMessageParams(converted_msg_params);
+  response_message[strings::msg_params] = converted_msg_params;
+  pending_requests_.pop_front();
+
+  // workaround for existing scripts that do not send response for each
+  // paticular data, but send only overal result
+  //  const auto successful_subscriptions =
+  //      SuccessfulSubscriptionsFromResponse(response_message);
```

Comment:
Don't forget to remove commented code

Reply:
renamed to pending_resumption_lock_ 
This si not a lock for some collection. 
This lock is to avoid simutanious on_event with resumption of other applicaiton and to make functions predictable